types.go

```diff
@@ -41,45 +41,67 @@ type digSentinel interface {
 	digSentinel()
 }
 
-// In is an embeddable object that signals to dig that the struct
-// should be treated differently. Instead of itself becoming an object
-// in the graph, memebers of the struct are inserted into the graph.
+// In may be embedded into structs to request dig to treat them as special
+// parameter structs. When a constructor accepts such a struct as an argument,
+// instead of the struct becoming a dependency for that constructor, all its
+// fields become dependencies instead. See the section on Parameter Objects
+// above for more information.
 //
-// Tags on those memebers control their behavior. For example,
-//
-//    type Input struct {
-//      dig.In
-//
-//      S *Something
-//      T *Thingy `optional:"true"`
-//    }
+// Fields of the struct may optionally be tagged to customize the behavior of
+// dig. The following tags are supported,
 //
+//   name        Requests a value with the same name and type from the
+//               container. See Named Values for more information.
+//   optional    If set to true, indicates that the dependency is optional and
+//               the constructor gracefully handles its absence.
 type In struct{ digSentinel }
 
 // Out is an embeddable type that signals to dig that the returned
 // struct should be treated differently. Instead of the struct itself
 // becoming part of the container, all members of the struct will.
-type Out struct{ digSentinel }
 
-// TODO: better usage docs
-// Try to add some symmetry for In-Out docs as well.
+// Out may be embedded into structs to request dig to treat them as special
```

Comment:
"special" is an adjective here, not a noun. There are two words that are part of this sentence on the following line.